docs/source/daml/reference/templates.rst

```diff
@@ -137,6 +137,31 @@ Choices
   Starting with ``choice`` lets you pass in a ``Party`` to use as a controller. But you must make sure to add that party as an ``observer``.
 - See :doc:`choices` for full reference information.
 
+Serializable Types
+******************
+
+Every parameter to a template, choice argument and choice result must have a *serializable type*.
+This does not merely mean "convertible to bytes"; it has a specific meaning in Daml.
+The serializability rule serves three purposes:
+
+1. offer a stable means to store ledger values permanently,
+2. provide a sensible encoding of them over :doc:`/app-dev/ledger-api`, and
+3. provide sensible *types* that directly match their Daml counterparts in languages like Java for language codegen.
+
+For example, certain kinds of type parameters Daml offers are perfectly compatible with (1) and (2), but have no proper counterpart in (3), so they are disallowed.
+Similarly, function types have a perfectly sensible Java counterpart, satisfying (3), but no reliable way to store or share them via the API, thus failing (1) and (2).
+
+The following types are *not serializable*, and thus may not be used in templates.
+
+1. function types
+2. record types where any field is not serializable
+3. variant types where any value is not serializable
+4. variant and enum types with no constructors
+5. references to a parameterized data type where any type argument is not serializable.
+   This applies whether or not the data type definition uses the type parameter
+6. defined data types with type parameters of kind ``Nat``, or any kind other than ``*``.
+   This means higher-kinded types and types that take a parameter just to pass to ``Numeric`` are not serializable
```

Comment:
```suggestion

- Function types.
- Record types where no field is serializable.
- Variant types where no value is serializable.
- Variant and enum types with no constructors.
- References to a parameterized data type where no type argument is serializable. This applies whether or not the data type definition uses the type parameter.
- Defined data types with type parameters of kind ``Nat``, or any kind other than ``*``. This means higher-kinded types, and types that take a parameter just to pass to ``Numeric``, are not serializable.
```

Reply:
> Record types where no field is serializable.

Would not be accurate; a record with one serializable field and one non-serializable field is non-serializable, despite not fitting the quoted change.

Reply:
Yeah, makes sense. I wonder if there's a clearer way of saying it, perhaps "where there is any non-serializable field"?

Reply:
I would definitely capitalize, add full stops, and put bullets instead of numbers.

Reply:
42754fb4e5d013d7c60fe56af22d1b56f7a48946